kibble/scanners/base.py

```diff
@@ -0,0 +1,31 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import logging
+from typing import Any
+
+
+class BaseScanner:
+    """Abstract, base class for all scanners"""
+
+    # pylint: disable=too-few-public-methods
+    def __init__(self, **kwargs):
+        self.log = logging.getLogger(__name__)
+
+    def _persist(self, payload: Any):  # pylint: disable=no-self-use
+        """Persists data to database. Should be implemented per scanner."""
```

Comment:
One scanner, one database/table?

Reply:
> One scanner, one database/table?

I assume there will be only one database with multiple collections / document types. Need to evaluate ES vs some  other nosql db

Reply:
as of current ES versions, you can't have different document types in the same DB.
So it would have to be one document type per index, which is what we also did with Kibble 1, for instance you have kibble_mail, kibble_code_commit, kibble_issue etc - one per general data type. You can do more generalized queries across indices, but usually we just deal with one type at a time, so having multiple indices is not an issue...